Userland/Libraries/LibJS/Runtime/RegExpPrototype.cpp

```diff
@@ -1075,6 +1099,23 @@ JS_DEFINE_NATIVE_FUNCTION(RegExpPrototype::compile)
         if (!flags.is_undefined())
             return vm.throw_completion<TypeError>(ErrorType::NotUndefined, flags.to_string_without_side_effects());
 
+        // https://github.com/tc39/proposal-regexp-legacy-features#regexpprototypecompile--pattern-flags-
+        // 3. Let thisRealm be the current Realm Record.
+        auto* this_realm = vm.current_realm();
+
+        // 4. Let oRealm be the value of O’s [[Realm]] internal slot.
+        auto* o_realm = regexp_object->vm().current_realm();
+
+        // 5. If SameValue(thisRealm, oRealm) is false, throw a TypeError exception.
+        if (this_realm != o_realm) {
+            return TypeError::create(*this_realm, "SameValue(thisRealm, oRealm) is false");
```

Comment:
Don't inline error strings, add an appropriate, user-facing error message to `ErrorTypes.h` instead (`SameValue` is an internal and nothing a user would understand)